### PR TITLE
feat(react-charting): Support changing legends programatically at runtime and bug fixes

### PR DESCRIPTION
**Pull request by AtishayMsft**
## New Behavior

- Add support to modify legends programatically at runtime.
- Security fixes for json depth and javascript brackets.
- Support propogating events from callout to the parent container.

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: change/@fluentui-react-charting-14c34b26-6050-4fa3-8ab9-ccfc1661f953.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-14c34b26-6050-4fa3-8ab9-ccfc1661f953.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33519&commitId=5765a28b660af7daf978240547623e3ea94cced6&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 5 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Bottom Left Edge RTL.chromium.png|2178| Changed|
|Callout.Top auto edge.chromium.png|2307| Changed|
|Callout.Bottom center.chromium.png|2146| Changed|
|Callout.Bottom Right Edge RTL.chromium.png|1593| Changed|
|Callout.Right bottom edge.chromium.png|3005| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.chromium.png|11| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-examples/src/react-charting/DeclarativeChart/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return { hasError: true, error: `${error.message} ${error.stack}` };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    console.error("Error caught by ErrorBoundary:", error, errorInfo);
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return <h1>${this.state.error}</h1>;
```
Comment (srmukher) [inline packages/react-examples/src/react-charting/DeclarativeChart/ErrorBoundary.tsx:29]: will we be showing the entire error stack on screen? instead can we console log the stack and show some message on screen?

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 3]
```diff
@@ -505,11 +504,32 @@ export const transformPlotlyJsonToGaugeProps = (
   };
 };
 
+const MAX_DEPTH = 8;
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:507]: how did we determine the max_depth of 8?

File: packages/react-examples/src/react-charting/DeclarativeChart/ErrorBoundary.tsx (hunk 2 above)
Comment (AtishayMsft) [reply packages/react-examples/src/react-charting/DeclarativeChart/ErrorBoundary.tsx:25]: console log is not recommended. This is a test example only. It is fine to show the error on the screen.

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts (hunk 3 above)
Comment (AtishayMsft) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:507]: Our data generally has a depth of 5/6. Adding 2 more levels to be on safe side.

**Review by srmukher — APPROVED** [review]

File: change/@fluentui-react-charting-14c34b26-6050-4fa3-8ab9-ccfc1661f953.json (hunk 1 above)
Comment (AtishayMsft) [reply change/@fluentui-react-charting-14c34b26-6050-4fa3-8ab9-ccfc1661f953.json:1]: VR unrelated to changes in this PR.

**Pull request merged**